### PR TITLE
16551 Add new eslint presets and update code accordingly

### PR DESCRIPTION
**Pull request by chenhongjing**
*Issue #:* /bcgov/entity#16551

*Description of changes:*
- Add new eslint presets and update code accordingly. New eslint presets are consistent with that in https://github.com/bcgov/business-filings-ui/pull/525
- Fixed errors that came up after running `npm run lint`
- App version: 5.3.0

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: tests/unit/ListPeopleAndRoles.spec.ts
Comment (chenhongjing) [inline tests/unit/ListPeopleAndRoles.spec.ts:359]: This fix the error occurred after `npm run lint`. It's the same as that in line 367.
image

File: package.json
Comment (severinbeauvais) [inline package.json:3]: As with 16552, this is a fairly significant change, so update the minor version (5.3.0).

**Comment by pwei1018** [comment]
Temporary Url for review: https://business-create-dev--pr-555-3wrjypm4.web.app

File: tests/unit/ListPeopleAndRoles.spec.ts
Comment (severinbeauvais) [reply tests/unit/ListPeopleAndRoles.spec.ts:359]: Yes, this is fine. It's because the text is on a newline, so the exact match fails.

File: src/components/Dissolution/CompleteResolution.vue
Comment (JazzarKarim) [inline src/components/Dissolution/CompleteResolution.vue:172]: Did SonarCloud complain about this?

File: src/components/Dissolution/CompleteResolution.vue
Comment (JazzarKarim) [inline src/components/Dissolution/CompleteResolution.vue:423]: Can you test this to make sure it's OK? It works fine for Filings UI and Edit UI so I guess it should be fine here also.

File: src/components/Dissolution/CompleteResolution.vue
Comment (severinbeauvais) [reply src/components/Dissolution/CompleteResolution.vue:172]: Looks like SonarCloud isn't running for this UI. @pwei1018 ?

**Comment by pwei1018** [comment]
Temporary Url for review: https://business-create-dev--pr-555-3wrjypm4.web.app

**Review by JazzarKarim — COMMENTED** [review]
So I went through every file. Looks OK to me!

File: src/components/Dissolution/CompleteResolution.vue
Comment (chenhongjing) [reply src/components/Dissolution/CompleteResolution.vue:423]: Yeah, I tested it and it works.

Screenshot 2023-06-21 120307

Also, you can find reference here that the shorthand of `v-slot` is `#`:
https://vuejs.org/guide/components/slots.html#named-slots

File: src/views/Dissolution/DissolutionReviewConfirm.vue
Comment (severinbeauvais) [reply src/views/Dissolution/DissolutionReviewConfirm.vue:128]: Fine by me.

File: src/components/Dissolution/CompleteResolution.vue
Comment (severinbeauvais) [reply src/components/Dissolution/CompleteResolution.vue:423]: ^^ Careful with the Vue docs -- some are for Vue3 and others are for Vue2.

File: src/components/Incorporation/IncorporationDateTime.vue
Comment (severinbeauvais) [inline src/components/Incorporation/IncorporationDateTime.vue:325]: 👍

File: src/components/Incorporation/ShareStructure.vue
Comment (severinbeauvais) [inline src/components/Incorporation/ShareStructure.vue:105]: This and line 102 are new replacements that I don't I've seen before. Please test both.

File: src/components/Incorporation/ShareStructure.vue
Comment (severinbeauvais) [inline src/components/Incorporation/ShareStructure.vue:154]: In case we develop anything like this in the future, there's no point in mixing strings and a function here -- it would be cleaner with just a function to return the whole label.

File: src/components/common/ListPeopleAndRoles.vue
Comment (severinbeauvais) [inline src/components/common/ListPeopleAndRoles.vue:68]: FYI, this is a Vuetify theme colour that we declared in main.ts. This is not a CSS variable as used elsewhere.

Comment (JazzarKarim) [reply src/components/common/ListPeopleAndRoles.vue:68]: So Sev from what I discovered, If we're using colors in the template, we use the ones declared in main.ts. If we're using colors in the the style block, we use the ones declared in the styles file. Right?

File: src/dialogs/FileAndPayInvalidNameRequestDialog.vue
Comment (severinbeauvais) [inline src/dialogs/FileAndPayInvalidNameRequestDialog.vue:13]: Since you're here, could you add a blank line above this (between title block and text block) please? Thanks!

File: src/components/Incorporation/ShareStructure.vue
Comment (chenhongjing) [reply src/components/Incorporation/ShareStructure.vue:105]: Just tested them. They work. And this usage can be found in section `Named Scoped Slots` at https://vuejs.org/guide/components/slots.html#named-slots

image

File: src/components/common/ListPeopleAndRoles.vue
Comment (severinbeauvais) [reply src/components/common/ListPeopleAndRoles.vue:68]: Close! If we're using colours in a Vuetify component then we use the ones in main.ts. If we use a class to set colours then we use the CSS variables.

Comment (JazzarKarim) [reply src/components/common/ListPeopleAndRoles.vue:68]: OK, noted. Thanks!

File: src/components/Incorporation/ShareStructure.vue
Comment (severinbeauvais) [reply src/components/Incorporation/ShareStructure.vue:105]: Thanks for testing and finding that reference. We get smarter every day :)

File: src/components/Dissolution/CompleteResolution.vue
Comment (JazzarKarim) [reply src/components/Dissolution/CompleteResolution.vue:423]: Great, thanks! Resolving.

File: src/components/common/ListPeopleAndRoles.vue
Comment (severinbeauvais) [reply src/components/common/ListPeopleAndRoles.vue:68]: Hongjng, you can resolve this comment after you've read it :)

**Comment by severinbeauvais** [comment]
Hongjing, I think you need to revert the `xxx.sync` to `v-model:xxx` changes. Do the same thing Karim did: turn the rule off and add a "FUTURE" comment so we know to enable the rule later (when we go to Vue/Vuetify 3).

I'll have one last look at the code changes before I approve.

File: src/components/Dissolution/CompleteResolution.vue
Comment (severinbeauvais) [reply src/components/Dissolution/CompleteResolution.vue:172]: Patrick, I think SonarCloud isn't running for Edit UI, either. Can you please enable for both UIs, or guide me through it? Thanks.

**Comment by chenhongjing** [comment]
> Hongjing, I think you need to revert the `xxx.sync` to `v-model:xxx` changes. Do the same thing Karim did: turn the rule off and add a "FUTURE" comment so we know to enable the rule later (when we go to Vue/Vuetify 3).
> 
> I'll have one last look at the code changes before I approve.

I've checked there's no such issue in Create UI, if I didn't miss any details. So there's no change, and I've just updated eslint presets.

File: src/components/Dissolution/CompleteResolution.vue
Comment (severinbeauvais) [reply src/components/Dissolution/CompleteResolution.vue:172]: This conversation doesn't block this PR. Patrick and I will look into this later when we both have a moment.

Karim, resolve?

**Review by severinbeauvais — APPROVED** [review]
Very good, thanks.

File: src/components/Dissolution/CompleteResolution.vue
Comment (JazzarKarim) [reply src/components/Dissolution/CompleteResolution.vue:172]: Agreed. Resolving.

**Review by JazzarKarim — APPROVED** [review]
Great job! 👍

**Pull request merged**
